clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -2606,6 +2606,19 @@ class UnsafeBufferUsageReporter : public UnsafeBufferUsageHandler {
 #endif
   }
 
+  void handleUnsafeUniquePtrArrayAccess(const DynTypedNode &Node,
+                                        bool IsRelatedToDecl,
+                                        ASTContext &Ctx) override {
+    SourceLocation Loc;
+    std::string Message;
+
+    Loc = Node.get<Stmt>()->getBeginLoc();
+    Message = "Direct operator[] access on std::unique_ptr<T[]> is unsafe "
```

Comment:
This message doesn't look that much different from what we put for `warn_unsafe_buffer_usage_unique_ptr_array_access` in `clang/include/clang/Basic/DiagnosticSemaKinds.td`.
Maybe we could keep just one of them?